contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -81,9 +83,55 @@ contract LidoOracle is ILidoOracle, AragonApp {
     bytes32 internal constant MIN_REPORTABLE_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.minReportableEpochId");
     /// @dev the max id of reported epochs
     bytes32 internal constant MAX_REPORTED_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.maxReportedEpochId");
+    /// @dev storage for the last completed report and its time
+    bytes32 internal constant LAST_COMPLETED_REPORT_TOTAL_SUPPLY = keccak256("lido.LidoOracle.lastCompletedReportTotalSupply");
+    bytes32 internal constant LAST_COMPLETED_REPORT_TIME = keccak256("lido.LidoOracle.lastCompletedReportTime");
+    bytes32 internal constant LAST_COMPLETED_REPORT_VALIDATORS = keccak256("lido.LidoOracle.lastCompletedReportValidators");
     /// @dev storage for all gathered from reports data
     mapping(uint256 => EpochData) private gatheredEpochData;
 
+    /*
+    If we use APR as a basic reference for increase, and expected range is between 1 and 10% APR.
+    So the granularity step for yearly APR is 1%
+    daily_granularity = 0.01 / 365 = 2.74e-05
+    Currently used basisPoints (1e-4) is unable to express such microscopic ratio.
+    So we should take another popular metric to express this - PPM or 1e-6
+    daily_granularity_in_ppm = 2.74e-05 / 1e-6 = 27 (pretty good)
+    with 100% buffer for future spikes it will be 50
+
+    may be changed by Gov
+    */
+    uint128 public ALLOWED_BEACON_BALANCE_INCREASE = 50;  // PPM
```

Comment:
We need to keep variables in unstructured storage (keccak256 shift, as above) to comply with upgradeability.